core/src/stack/inputs.rs

```diff
@@ -77,21 +77,27 @@ impl ToElements<Felt> for StackInputs {
 
 impl Serializable for StackInputs {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
-        // TODO the length of the stack, by design, will not be greater than `u32::MAX`. however,
+        // TODO the length of the stack, by design, will not be greater than `u16::MAX`. however,
         // we must define a common serialization format as we might diverge from the implementation
         // here and the one provided by default from winterfell.
 
```

Comment:
The comment is incorrect here - the depth of the stack can get more than `u16:MAX`. Also, looking at the comment, we should probably introduce a constant for this. Maybe something like `MAX_INPUT_STACK_DEPTH`.

Reply:
We should also probably enforce this limit in the constructor.